doc/manual/source/store/building.md

```diff
@@ -1,101 +1,187 @@
+
+> **Explanation**
+>
+> Ultimately, what matters for determinism is the behavior of IO operations that the process attempts (whether these are successes or failures), because of how they affect the output files, and how they affect the further execution of the builder process.
+> From Nix (and the operating system)'s perspective, there are many, many different ways --- different implementation strategies --- of effecting the same I/O behavior,
+> But from the process's perspective, there is only one correct behavior.
+
+## What derivations can be built
+
+Actually only some derivations are ready to be built.
+In particular, only [*resolved*](./resolution.md) derivations can be built.
+That is to say, a derivation that depends on other derivations is not ready yet to be built, because those other derivations might not be built.
+If the other derivations are indeed built, we can witness this fact by resolving the derivation, and converting all the derivation's input references into plain store paths.
+
+> **Note**
+>
+> Note that [input-addressing](derivation/outputs/input-address.md) derivations are improperly resolved.
+> As discussed on the linked page, the current input-addressing algorithm does not respect resolution-equivalence of derivations (\\(\\sim_\mathrm{Drv}\\)).
+> That means that if Nix properly resolved an input-addressed derivation, the resolved derivation would have different input addresses, violating expectations.
+> Nix therefore improperly resolves the derivation, keeping its original input address output paths, creating an invalid derivation that is both resolved and instructed to create the outputs at the originally expected paths.
+
+## Environment of the builder process
 
 The [`builder`](./derivation/index.md#builder) is executed as follows:
 
-- A temporary directory is created where the build will take place. The
-  current directory is changed to this directory.
+### File system
+
+The builder should have access to a limited file system where only certain objects are available.
+The most important exposed files are the inputs (other store objects) of the (resolved) derivation.
+Additionally, some other files are exposed.
+
+#### Store inputs
+
+The builder will be run against a file system in which the [closure] of the inputs is mounted inside the [store directory][store directory path].
+In particular, consider a store that just contains this closure.
+That store may be exposed to the file system according to the rules specified in the [Exposing Store Objects in OS File Systems](./store-path.md#exposing) documentation.
+This precisely defines the file system layout of the store that should be visible to the builder process.
+
+> **Note**
+>
+> Historically, Nix exposed *at least* the following store contents to the builder, but also arbitrarily other store objects, due to limitations around operating systems' file system virtualization capabilities, and wanting to avoid copying or moving files.
```

Comment:
We shouldn't use the word "Historically" if it's still the case today. We're not documenting platonic ideal Nix :-)